### PR TITLE
feat: optimizations of checks

### PR DESCRIPTION
**Pull request by xuzhu-591**
### Description of your changes
* add `Ready` api instead of `ForceExecute` api, which is used to mark pipelinerun as ready status forcefully.
* support root members.
* support system message about pipelinerun, such as "tom executed the pipelinerun".

Fixes #

I have:

- [x] Read and followed Horizon's [contribution process](https://github.com/horizoncd/horizon/CONTRIBUTING.md).
- [x] Run `make build && make lint` to ensure this PR is ready for review.

### CODE REVIEW

**Comment by work-jlsun** [comment]
just set a chat message

**Comment by xuzhu-591** [comment]
test

File: core/controller/pipelinerun/controller.go [hunk 1]
```diff
@@ -310,7 +312,12 @@ func (c *controller) StopPipelinerun(ctx context.Context, pipelinerunID uint) (e
 		return errors.E(op, err)
 	}
 
-	return tektonClient.StopPipelineRun(ctx, pipelinerun.CIEventID)
+	err = tektonClient.StopPipelineRun(ctx, pipelinerun.CIEventID)
+	if err != nil {
+		return err
+	}
+	c.prSvc.CreateSystemMessage(ctx, pipelinerunID, "stopped the pipelinerun")
```
Comment (kilosonc) [inline core/controller/pipelinerun/controller.go:319]: do not use literal string

File: core/controller/pipelinerun/controller.go [hunk 2]
```diff
@@ -338,7 +345,12 @@ func (c *controller) StopPipelinerunForCluster(ctx context.Context, clusterID ui
 		return errors.E(op, err)
 	}
 
-	return tektonClient.StopPipelineRun(ctx, pipelinerun.CIEventID)
+	err = tektonClient.StopPipelineRun(ctx, pipelinerun.CIEventID)
+	if err != nil {
+		return err
+	}
+	c.prSvc.CreateSystemMessage(ctx, pipelinerun.ID, "stopped the pipelinerun")
```
Comment (kilosonc) [inline core/controller/pipelinerun/controller.go:352]: same

File: core/controller/pipelinerun/controller.go [hunk 3]
```diff
@@ -390,6 +402,30 @@ func (c *controller) Execute(ctx context.Context, pipelinerunID uint, force bool
 	}
 	c.eventSvc.CreateEventIgnoreError(ctx, common.ResourcePipelinerun, pipelinerunID,
 		eventmodels.PipelinerunExecuted, nil)
+	if force {
+		c.prSvc.CreateSystemMessage(ctx, pipelinerunID, "forced to execute the pipelinerun")
+	} else {
+		c.prSvc.CreateSystemMessage(ctx, pipelinerunID, "executed the pipelinerun")
```
Comment (kilosonc) [inline core/controller/pipelinerun/controller.go:408]: same

File: core/controller/pipelinerun/controller.go [hunk 4]
```diff
@@ -390,6 +402,30 @@ func (c *controller) Execute(ctx context.Context, pipelinerunID uint, force bool
...
+	}
+	return nil
+}
+
+func (c *controller) Ready(ctx context.Context, pipelinerunID uint) error {
```
Comment (kilosonc) [inline core/controller/pipelinerun/controller.go:413]: What is this method for?

File: core/controller/pipelinerun/controller.go [hunk 5]
```diff
@@ -390,6 +402,30 @@ func (c *controller) Execute(ctx context.Context, pipelinerunID uint, force bool
...
+		return err
+	}
+	if pr.Status != string(prmodels.StatusPending) {
+		return perror.Wrapf(herrors.ErrParamInvalid, "pipelinerun is not pending")
+	}
+	err = c.prMgr.PipelineRun.UpdateStatusByID(ctx, pipelinerunID, prmodels.StatusReady)
+	if err != nil {
+		return err
+	}
+	c.prSvc.CreateSystemMessage(ctx, pipelinerunID, "marked the pipelinerun as ready")
```
Comment (kilosonc) [inline core/controller/pipelinerun/controller.go:428]: no literal string

File: core/controller/pipelinerun/controller.go [hunk 6]
```diff
@@ -672,6 +708,7 @@ func (c *controller) Cancel(ctx context.Context, pipelinerunID uint) error {
 	}
 	c.eventSvc.CreateEventIgnoreError(ctx, common.ResourcePipelinerun, pipelinerunID,
 		eventmodels.PipelinerunCancelled, nil)
+	c.prSvc.CreateSystemMessage(ctx, pipelinerunID, "cancelled the pipelinerun")
```
Comment (kilosonc) [inline core/controller/pipelinerun/controller.go:711]: same

File: core/controller/pipelinerun/controller.go (hunk 1 above)
Comment (kilosonc) [reply core/controller/pipelinerun/controller.go:319]: can we make this `CreateMessage` async?

**Review by kilosonc — APPROVED** [review]

**Pull request merged**